File: utility_test.ts
import { assertEquals, assertNotEquals } from "@std/assert";
import { getObject, verifyIdentifier } from "./utility.ts";

function noOp(_s?: string): boolean {
  return true;
}

Deno.test("test verifyIdentifier", async function () {
  const object_urls: string[] = [
    "https://caltechlibrary.github.io/dataset/README.md",
  ];

  for (const url of object_urls) {
    const res = await verifyIdentifier(url, url, noOp);
    assertEquals(res, true);
  }
});

Deno.test("test getObject", async function () {
  const object_urls: string[] = [
    "https://feeds.library.caltech.edu/recent/article.json",
  ];
  for (const url of object_urls) {
    const res = await getObject(url, url, noOp);
    //console.log(`DEBUG res -> ${res}`);
    assertNotEquals(res, undefined);
  }
});

File: utility_test.go
package metadatatools

import (
	"testing"
)

func noOp(_ string) bool {
	return true
}

func TestVerifyIdentifier(t *testing.T) {
	if testing.Short() {
		t.Skip("skipping network test in short mode")
	}
	objectURLs := []string{
		"https://caltechlibrary.github.io/dataset/README.md",
	}
	for _, u := range objectURLs {
		if ok := VerifyIdentifier(u, u, noOp); !ok {
			t.Errorf("VerifyIdentifier(%q, %q, noOp) = false, expected true", u, u)
		}
	}
}

func TestGetObject(t *testing.T) {
	if testing.Short() {
		t.Skip("skipping network test in short mode")
	}
	objectURLs := []string{
		"https://feeds.library.caltech.edu/recent/article.json",
	}
	for _, u := range objectURLs {
		obj, ok := GetObject(u, u, noOp)
		if !ok || obj == nil {
			t.Errorf("GetObject(%q, %q, noOp) = (%v, %v), expected an object and true", u, u, obj, ok)
		}
	}
}
